apps/mobile/src/features/send-refactor/forms/btc/btc-form.tsx

```diff
@@ -0,0 +1,153 @@
+import { type Account } from '@/store/accounts/accounts';
+import { whenInputCurrencyMode } from '@/utils/when-currency-input-mode';
+import { t } from '@lingui/macro';
+
+import {
+  type AverageBitcoinFeeRates,
+  type FiatCurrency,
+  type MarketData,
+  type Money,
+} from '@leather.io/models';
+import { type Utxo } from '@leather.io/query';
+import { BtcAvatarIcon, Button } from '@leather.io/ui/native';
+import { isNumber } from '@leather.io/utils';
+
+const protocol = 'nativeBtc';
+const symbol = 'BTC';
+
+interface BtcFormProps {
+  account: Account;
+  availableBalance: Money;
+  fiatBalance: Money;
+  feeRates: AverageBitcoinFeeRates;
+  utxos: Utxo[];
+  marketData: MarketData;
+  fiatCurrency: FiatCurrency;
+  assetItemAnimationOffsetTop?: number | null;
+  onOpenAssetPicker(): void;
+}
+
+export function BtcForm({
+  fiatBalance,
+  availableBalance,
+  marketData,
+  account,
+  utxos,
+  feeRates,
+  fiatCurrency,
+  onOpenAssetPicker,
+  assetItemAnimationOffsetTop,
+}: BtcFormProps) {
+  const {
+    state: { inputCurrencyMode },
+  } = useSendFlowContext();
+  const currency = whenInputCurrencyMode(inputCurrencyMode)({
+    crypto: symbol,
+    fiat: fiatCurrency,
+  });
+  const { form, maxSpend, onSetMax, onSubmit } = useBtcForm({ account, feeRates, utxos });
+
+  return (
+    <SendFormContainer>
+      <AssetDisplay
+        availableBalance={availableBalance}
+        protocol={protocol}
+        fiatBalance={fiatBalance}
+        name={t({
+          id: 'asset_name.bitcoin',
+          message: 'Bitcoin',
+        })}
+        symbol={symbol}
+        icon={<BtcAvatarIcon />}
+        onPress={onOpenAssetPicker}
+        assetItemElementInitialOffset={assetItemAnimationOffsetTop ?? null}
+      />
+      <Controller
+        render={({ field: { value }, fieldState: { invalid, isValidating } }) => (
+          <AmountField
+            inputValue={value}
+            cryptoCurrency={symbol}
+            fiatCurrency={fiatCurrency}
+            marketData={marketData}
+            invalid={invalid}
+            isValidating={isValidating}
+            inputCurrencyMode={inputCurrencyMode}
+            enteringAnimationEnabled={isNumber(assetItemAnimationOffsetTop)}
+            onSetIsSendingMax={() => onSetMax(true)}
+            canSendMax={true}
+            locale={locale}
+          />
+        )}
+        control={form.control}
+        name="amount"
+      />
+
+      <Controller
+        control={form.control}
+        name="recipient"
+        render={({
+          field: { value, onChange, onBlur },
+          fieldState: { invalid, isDirty, isTouched, error },
+        }) => (
+          <Recipient
+            value={value}
+            onChange={onChange}
+            onBlur={onBlur}
+            invalid={invalid}
+            isDirty={isDirty}
+            isTouched={isTouched}
+            error={error}
+          />
+        )}
+      />
+
+      {!form.formState.isValidating ? (
```

Comment:
Should this also have a check to only render if `isValidating` and `isError`?

Reply:
Other displaying heuristics are baked into the Component, there's no reason why `isValidating` shouldn't be too. Thanks, fixed.